mycpp/pass_state.py

```diff
@@ -18,17 +18,17 @@ class Virtual(object):
   """
 
     def __init__(self) -> None:
-        self.methods: dict[str, list[str]] = defaultdict(list)
-        self.subclasses: dict[str, list[str]] = defaultdict(list)
-        self.virtuals: list[tuple[str, str]] = []
-        self.has_vtable: dict[str, bool] = {}
-        self.can_reorder_fields: dict[str, bool] = {}
+        self.methods: dict[tuple[str], list[str]] = defaultdict(list)
+        self.subclasses: dict[tuple[str], list[tuple[str]]] = defaultdict(list)
+        self.virtuals: dict[tuple[tuple[str], str], Optional[tuple[str, str]]] = {}
```

Comment:
It might be nice to define a type alias `SymbolName = tuple[str]` or `SymbolPath`

Reply:
Good idea. Will add this